reference/halyard/secrets/index.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: single
+title:  "Secrets"
+sidebar:
+  nav: reference
+---
+
+{% include toc %}
+
+Managing Spinnaker secrets separately from its configuration is a necessary step to enabling Spinnaker through an SCM like git. Spinnaker supports end-to-end secrets management, starting in version 1.14. Simply replace secrets in the Halconfig and service profiles with the syntax described here and Spinnaker will decrypt them as needed. 
+
+
+### Secret Format
+When referencing secrets in configs, we use the following general format:
+
+```
+encrypted:<secret engine>!<key1>:<value1>!<key2>:<value2>!...
+```
+The key-value parameters making up the string vary with each secret engine. Refer to the specific documentation for each engine for more information.
+
+### In Halyard
```

Comment:
Yeah might be worth showing examples of the difference in the`hal` commands i.e. instead of setting the value as the plain text secret, you set it with the encrypted syntax.

Reply:
Unfortunately `hal` commands don't really work. It can't handle the `!` in the syntax if it's all in one command. It does work for commands where it prompts you for the secret after entering the command. 

Fails:   `$ hal config storage s3 edit --secret-access-key encrypted:s3!r:us-west-2!b:mybucket!k:key`
Works: `$ hal config storage s3 edit --secret-access-key`
`Your AWS Secret Key.:` (enter `encrypted:vault!r:us-west-2!b:mybucket!k:key` when prompted here)

Reply:
Ahh! Might be worth pointing out in this docs then that using it via a hal command would not work :)

Reply:
Ok, will do!